### PR TITLE
Keep the annotation "pv.kubernetes.io/provisioned-by" when restoring PVs

### LINKED ISSUE #3470: PV is not getting cleaned up for the restored volumes
**What steps did you take and what happened:**

Restored the namespace which has 2 pods and corresponding pvcs using openebs/velero-plugin. When I am deleting the restored namespace, pv remains there in the system with the error 

```
  status:
    message: 'Error getting deleter volume plugin for volume "pvc-b4b70408-49d7-4921-a88e-80c1fcd981aa":
      no deletable volume plugin matched'
    phase: Failed

```

**What did you expect to happen:**

PV to be cleand up. PVC and pods gets cleaned up but not the PV.

**Anything else you would like to add:**

I think this change (https://github.com/vmware-tanzu/velero/pull/3007) caused the issue. k8s checks the annotation `pv.kubernetes.io/provisioned-by` to find the plugin related information which has been cleaned up in this PR. Now k8s assumes that pv is statically provisioned and throws error.

This works fine with 1.5.1 and lower releases. The issue is there with 1.5.2 and 1.5.3.

### PR DESCRIPTION
**Pull request by ywk253100**
More details please refer to https://github.com/vmware-tanzu/velero/issues/3470#issuecomment-976279606

Fixes #3470

Signed-off-by: Wenkai Yin(尹文开) <yinw@vmware.com>

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required`.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: changelogs/unreleased/4391-ywk253100 [hunk 1]
```diff
@@ -0,0 +1 @@
+Keep the annotation "pv.kubernetes.io/provisioned-by" when storing PVs
```
Comment (jenting) [inline changelogs/unreleased/4391-ywk253100:1]:
```suggestion
Keep the annotation "pv.kubernetes.io/provisioned-by" when restoring PVs
```

**Comment by ywk253100** [comment]
Tested on:
- AKS
  - `v1.19.11`: only in-tree drivers are installed
  - `v1.21.2`: both in-tree and CSI drivers are installed, and the CSI SC is the default storage class
- EKS
  - `v1.18.20-eks-8c579e`: only in-tree drivers are installed
  - `v1.20.7-eks-d88609`: only in-tree drivers are installed
- vSphere
  - `v1.21.2+vmware.1`: only CSI drivers are installed

**Review by reasonerjt — DISMISSED** [review]

File: pkg/restore/restore_test.go [hunk 2]
```diff
@@ -2989,15 +2989,15 @@ func Test_resetVolumeBindingInfo(t *testing.T) {
 				WithName("pv-1").WithAnnotations(
 				KubeAnnBindCompleted,
 				KubeAnnBoundByController,
-				KubeAnnDynamicallyProvisioned,
```
Comment (jenting) [inline pkg/restore/restore_test.go:2992]: I don't know why need this change that removing the const variable and use string instead.

**Review by jenting — APPROVED** [review]
LGTM, thank you.

**Review by reasonerjt — APPROVED** [review]

**Pull request merged**